packages/forms-web-app/src/dynamic-forms/question.js

```diff
@@ -1,69 +1,21 @@
-/***********************************************************
- * This file holds the class definitions for the different *
- * components (question types) including base types        *
- ***********************************************************/
-
-const nunjucks = require('nunjucks');
-
-// todo:
-// test
-// refactor
-// jsdoc
-
 class Question {
 	title;
 	question;
 	description;
 	type;
 	fieldName;
+	validators = [];
 
 	constructor({ title, question, description, type, fieldName } = {}) {
+		if (!title || title === '') throw new Error('title parameter is mandatory');
+		if (!question || question === '') throw new Error('question parameter is mandatory');
+		if (!fieldName || fieldName === '') throw new Error('fieldName parameter is mandatory');
 		this.title = title;
 		this.question = question;
 		this.description = description;
 		this.type = type;
 		this.fieldName = fieldName;
 	}
-
-	prepQuestionForRendering(answers) {
```

Comment:
The `prepQuestionForRendering` method is used in the controller so that shouldn't be removed. Arguably the part relating to options could be moved to an override in the OptionsQuestion class as it would only be needed on a question with options.

Reply:
Can you update the ticket?